ts/screens/modal/UpdateAppModal.tsx

```diff
@@ -63,31 +62,41 @@ class UpdateAppModal extends React.PureComponent<never, State> {
   }
 
   public componentWillUnmount() {
-    if (this.idTimeout) {
-      clearTimeout(this.idTimeout);
-    }
     BackHandler.removeEventListener("hardwareBackPress", this.handleBackPress);
   }
 
+  private setError = () =>
+    new Promise(resolve => {
+      this.setState(
+        {
+          hasError: true
+        },
+        resolve
+      );
+    });
+
   private openAppStore = () => {
-    // the error is already displayed
     if (this.state.hasError) {
       return;
     }
-    // storeUrl is not a webUrl, try to open it
-    Linking.openURL(storeUrl).catch(() => {
-      // Change state to show the error message
-      this.setState({
-        hasError: true
-      });
-      // After 5 seconds restore state
-      // eslint-disable-next-line
-      this.idTimeout = setTimeout(() => {
-        this.setState({
-          hasError: false
-        });
-      }, timeoutErrorMsg);
-    });
+
+    // Play/App store native URL
+    // eslint-disable-next-line
```

Comment:
We may avoid to disable linting here just rework below lines of code as follows:

```typescript
    // Play/App store native URL
    Linking.openURL(storeUrl)
      // Try to fallback to the web URL
      .catch(() => Linking.openURL(webStoreURL))
      // No URL could be opened, show an error message
      .catch(() =>
        this.setError()
          // Hide the error after 5 seconds
          .then(() =>
            setTimeout(
              () =>
                this.setState({
                  hasError: false
                }),
              timeoutErrorMsg
            )
          )
      );
```